test/jest/victory-voronoi/victory-voronoi.test.js

```diff
@@ -0,0 +1,201 @@
+      const data = [
+        { x: 1, y: 2 },
+        { x: null, y: 4 },
+        { x: 5, y: null },
+        { x: 1, y: 2 }
+      ];
+      const { container } = render(<VictoryVoronoi data={data} />);
+      expect(container.querySelectorAll("path")).toHaveLength(2);
+    });
+  });
+
+  describe("event handling", () => {
+    it("attaches an event to the parent svg", () => {
+      const clickHandler = jest.fn();
+      const { container } = render(
+        <VictoryVoronoi
+          events={[
+            {
+              target: "parent",
+              eventHandlers: { onClick: clickHandler }
+            }
+          ]}
+        />
+      );
+      const svg = container.querySelector("svg");
+      fireEvent.click(svg);
+      expect(clickHandler).toBeCalled();
+      // the first argument is the standard evt object
+      expect(Object.keys(clickHandler.mock.calls[0][1])).toEqual(
+        expect.arrayContaining(["data", "scale", "width", "height", "style"])
+      );
+    });
+
+    it("attaches an event to data", () => {
+      const clickHandler = jest.fn();
+      const { container } = render(
+        <VictoryVoronoi
+          events={[
+            {
+              target: "data",
+              eventHandlers: { onClick: clickHandler }
+            }
+          ]}
+        />
+      );
+      const data = container.querySelectorAll("path");
+      [...data].forEach((node, index) => {
+        clickHandler.mockClear();
+
+        fireEvent.click(node);
+
+        expect(clickHandler).toBeCalled();
+        // the first argument is the standard evt object
+        const [, evProps, evIndex] = clickHandler.mock.calls[0];
+        expect(evProps).toMatchObject({ id: `voronoi-data-${index}` });
+        expect(evIndex).toEqual(`${index}`);
+      });
+    });
+
+    it("attaches an event to a label", () => {
+      const clickHandler = jest.fn();
+      const { container } = render(
+        <VictoryVoronoi
+          label="okay"
+          events={[
+            {
+              target: "labels",
+              eventHandlers: { onClick: clickHandler }
+            }
+          ]}
+        />
+      );
+      const labels = container.querySelectorAll("text");
+      [...labels].forEach((node, index) => {
+        clickHandler.mockClear();
+
+        fireEvent.click(node);
+
+        expect(clickHandler).toBeCalled();
+        // the first argument is the standard evt object
+        const [, evProps, evIndex] = clickHandler.mock.calls[0];
+        expect(evProps).toMatchObject({ text: "okay" });
+        expect(evIndex).toEqual(`${index}`);
+      });
+    });
+  });
+
+  describe("accessibility", () => {
+    it("adds an aria role to the path area", () => {
+      const { container } = render(<VictoryVoronoi />);
+      container.querySelectorAll("path").forEach((p) => {
+        const roleValue = p.getAttribute("role");
+        expect(roleValue).toEqual("presentation");
+      });
```

Comment:
Nicpick: We can replace this `forEach` with something like `expect(screen.getByRole('presentation')).toHaveLength(<expected path length>)`

Reply:
I added `toHaveLength` assertions for all `querySelectorAll` queries
Surprisingly, one of the "labels" test was returning 0 labels.  This was also happening on the old test, so I didn't fix anything yet.  Maybe we need to fix this?

Reply:
Hmm, strange! Let's leave it for now with a comment to investigate more when we refactor.